### PR TITLE
Solve GPU reference issues

### PR DESCRIPTION
**Pull request by hXl3s**
This MR fixes issues:
https://github.com/ZhiyuLi-goog/MoE_study/issues/13 - Weight decay is now correctly propagated
https://github.com/ZhiyuLi-goog/MoE_study/issues/11 - Not Applicable, tokenizer is relative to checkpoint path, not hardcoded
https://github.com/ZhiyuLi-goog/MoE_study/issues/10 - 8 is device per node, it is fine to assume it is constant, but now it is decided at runtime with torch.cuda.device_count()
https://github.com/ZhiyuLi-goog/MoE_study/issues/9 - Should be fixed now, scripts required manually putting validation subset into <workdir>/raw/

### CODE REVIEW

**Comment by hXl3s** [comment]
Rollback container to nemo:dev (it is an old tag, that is more stable than released one in case of MoE)

File: MoE_language_model/scripts/gpu/dataset_preprocessing.py [hunk 1]
```diff
@@ -52,15 +51,12 @@ def merge_files(output_path: pathlib.Path, input_paths: list[pathlib.Path]):
                     file_content = input_file.read()
                 output_file.write(file_content)
 
-    for i in range(8):
+    for i in range(6, 8):
         file_chunks = [
             source_directory / training_files[j] for j in range(i * 128, (i + 1) * 128)
```
Comment (ZhiyuLi-goog) [inline MoE_language_model/scripts/gpu/dataset_preprocessing.py]: Emm. If not mistaken, we may still see the same issue

```
training_files = [f"en/c4-train.{i:05d}-of-01024.json.gz" for i in range(768, 1024)]
# indicate len(training_files) = 128 

    for i in range(6, 8):
        file_chunks = [
            source_directory / training_files[j] for j in range(i * 128, (i + 1) * 128)
# say i = 6 range would be (768, xxx)
# and we still see the same out of range issue in training_files[768]
```

**Review by ZhiyuLi-goog — APPROVED** [review]
LGTM except the above comments.

File: MoE_language_model/scripts/gpu/dataset_preprocessing.py (hunk 1 above)
Comment (hXl3s) [reply MoE_language_model/scripts/gpu/dataset_preprocessing.py]: 6 would be 768-896
7 would be 896-1024

It should be correct?

Comment (ZhiyuLi-goog) [reply MoE_language_model/scripts/gpu/dataset_preprocessing.py]: j=768 in training_files[j]  would be out of range, since len(training_files) is 128

how about change
```
- training_files = [f"en/c4-train.{i:05d}-of-01024.json.gz" for i in range(768, 1024)]
+ training_files = [f"en/c4-train.{i:05d}-of-01024.json.gz" for i in range(1024)]
```
Any conflicts elsewhere?

Comment (hXl3s) [reply MoE_language_model/scripts/gpu/dataset_preprocessing.py]: you are right, I unnecessarily changed it before committing
